netmanager/src/views/pages/Download/Download.js

```diff
@@ -184,9 +184,94 @@ const Download = (props) => {
           link.remove();
         }
       })
-      .catch((err) => console.log(err && err.response && err.response.data));
+      .catch((err) => {
+        console.log(err && err.response && err.response.data);
+        dispatch(
+          updateMainAlert({
+            show: true,
+            message: "Unable to fetch data. Please try again",
+            severity: "error",
+          })
+        );
+      });
     setLoading(false);
   };
+
+  const handleUrbanBetterFormSubmit = async (e) => {
+    e.preventDefault();
```

Comment:
I am about to merge this one in since it is just an addition of a new function. I do not foresee any problems. @123MwanjeMike @Codebmk @FaithDaka